rpc/doc.go

```diff
@@ -0,0 +1,61 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+// Overall sequence flow
+
+// 1. An API facade is registered with the root object's registry using a method such as:
+//    registry.MustRegister("ModelManager", 10, ...)
+// 2. When a user executes a command (e.g., AddModelCommand), an API request containing
+//    a struct from the params package is sent to the API client with the appropriate API facade.
+// 3. The RPC client sends the facade name, version number, and parameter struct as part of its request to the RPC server.
+// 4. The RPC server finds the appropriate rpc method using the facade name and version number
+//    and executes the corresponding API function using conn.runRequest.
+
+// Related packages
+
+// - Codec: Defines methods for reading and writing RPC messages, enabling serialization and deserialization.
+// - Params: Defines the struct for json RPC requests
```

Comment:
This info feels like it could be rolled into the intro paragraph: You say what this package does and then how  it relates to these other packages.

Reply:
sure, thanks teodora